File: fixtures/java/tink/hmac-sha256/Main.java
import com.google.crypto.tink.Mac;
import com.google.crypto.tink.KeysetHandle;
import com.google.crypto.tink.mac.MacConfig;
import com.google.crypto.tink.mac.MacKeyTemplates;

public class Main {
    public static void main(String[] args) throws Exception {
        MacConfig.register();
        
        // Generate key
        KeysetHandle keysetHandle = KeysetHandle.generateNew(
            MacKeyTemplates.HMAC_SHA256_256BITTAG);
        
        // Get primitive
        Mac mac = keysetHandle.getPrimitive(Mac.class);
        
        byte[] message = "Hello, World!".getBytes();
        
        // Create MAC
        byte[] tag = mac.computeMac(message);
        
        // Verify MAC
        mac.verifyMac(tag, message);
    }
}

File: fixtures/python/tink/hmac-sha256/main.py
import tink
from tink import mac

# Register primitives
mac.register()

# Generate key
keyset_handle = tink.new_keyset_handle(mac.mac_key_templates.HMAC_SHA256_256BITTAG)

# Get primitive
mac_primitive = keyset_handle.primitive(mac.Mac)

message = b"Hello, World!"

# Create MAC
tag = mac_primitive.compute_mac(message)

# Verify MAC
mac_primitive.verify_mac(tag, message)
